src/Malinina.Homework.TaskМanager/Calendar.cs

```diff
@@ -0,0 +1,192 @@
+        {
+            if (GetTask(out Task task))
+            {
+                m_tasks.Remove(task);
+            }
+        }
+
+        /// <summary>
+        /// Display of all tasks.
+        /// </summary>
+        public void ShowAllTasks()
+        {
+            if (m_tasks.Count == 0)
+            {
+                Console.WriteLine("Calendar is empty");
+                return;
+            }
+            for (int i = 0; i < m_tasks.Count; i++)
+            {
+                Console.WriteLine($"ID: {m_tasks[i].Id}");
+                Console.WriteLine($"Description: {m_tasks[i].Description}");
+                Console.WriteLine($"Start date: {m_tasks[i].StartDate}");
+                Console.WriteLine($"Deadline: {m_tasks[i].Deadline}");
+                Console.WriteLine($"Status: {m_tasks[i].Status}");
+                Console.WriteLine("--------------------------------------------");
+            }
+        }
+
+        /// <summary>
+        /// Checking the existence of a task.
+        /// </summary>
+        /// <param name="task">Task.</param>
+        /// <returns>true or false.</returns>
+        private bool GetTask(out Task task)
+        {
+            task = null;
+
+            if (m_tasks.Count == 0)
+            {
+                Console.WriteLine("Calendar is empty");
+                return false;
+            }
+
+            Console.WriteLine("Enter task ID: ");
+            string inputeId = Console.ReadLine();
+
+            for (int i = 0; i < m_tasks.Count; i++)
+            {
+                if (m_tasks[i].Id == inputeId)
+                {
+                    task = m_tasks[i];
+                    return true;
+                }
+            }
+
+            Console.WriteLine("The task ID does not exist!");
+            return false;
+        }
+
+        /// <summary>
+        /// Processing the entered deadline.
+        /// </summary>
+        /// <param name="startTime">The entered date.</param>
+        /// <returns>Daedline is longer than the start date.</returns>
+        private DateTime GetDeadlineDate(DateTime startTime)
+        {
+            DateTime deadline = GetDate();
+
+            if(!CompareDates(startTime, deadline))
+            {
+                Console.WriteLine("Deadline can't be less then start date. Enter date again:");             
+                return GetDeadlineDate(startTime);
+            }
+
+            return deadline;
+        }
+
+        /// <summary>
+        /// Checking the format of the entered date.
+        /// </summary>
+        /// <returns>Correct date.</returns>
+        private DateTime GetDate()
+        {
+            string inputValue = Console.ReadLine();
+
+            if (DateTime.TryParse(inputValue, out DateTime date))
+            {
+                return date;
+            }
+            else
+            {
+                Console.WriteLine("Incorrect date! Enter again: ");
+                return GetDate();
```

Comment:
Please think what can be recursion side effects (when wrong date is entered a lot of times)